src/app/preview-area/preview-area.component.ts

```diff
@@ -0,0 +1,20 @@
+import {Component} from '@angular/core';
+import {Observable} from 'rxjs';
+import {AdAuthoringService} from '../ad-authoring/ad-authoring.service';
+import {map} from 'rxjs/operators';
+
+@Component({
+  selector: 'app-preview-area',
+  templateUrl: './preview-area.component.html',
+  styleUrls: ['./preview-area.component.scss'],
+})
+export class PreviewAreaComponent {
+  ampHtmlObs: Observable<ReadonlyArray<string>>;
+
+  constructor(private service: AdAuthoringService) {
+    // AmpHtml must be mapped into an array in order to use an aync pipe in the template
```

Comment:
Let's update this comment. The real reason we are mapping to an array is so the ngFor directive will remove the old preview and construct a new preview component on state changes. We should also add why this is necessary (We need to construct a new IFrame with the updated AMPHTML on state updates).